### PR TITLE
fix: review QA

### PR DESCRIPTION
**Pull request by rebelchris**
Feedback round from QA session.

- New blocked tag layout
- Passing from the sidebar down
- Empty tag state filter

DD-299 #done

### CODE REVIEW

**Comment by rebelchris** [comment]
@idoshamun If you have time, could you check this refactor work? 🤗

File: packages/shared/src/components/Sidebar.tsx [hunk 1]
```diff
@@ -17,6 +21,19 @@ export default function Sidebar(): ReactElement {
...
+    action?: () => unknown;
+  }>();
+
+  const initUnblockModal = ({
+    tag = null,
+    source = null,
+    action,
+  }: UnblockModalType) => {
+    setUnblockItem({ tag, source, action });
+  };
```
Comment (idoshamun) [inline packages/shared/src/components/Sidebar.tsx:36]: Isn't it redundant? Why don't you use `setUnblockItem` directly?

File: packages/shared/src/components/filters/CategoryButton.tsx [hunk 2]
```diff
@@ -28,8 +28,8 @@ export default function CategoryButton({
 }: {
   category: TagCategory;
   followedTags?: Array<string>;
-  onFollowTags: (tags, category?) => void;
-  onUnfollowTags: (tags, category?) => void;
+  onFollowTags?: (tags, category?) => void;
+  onUnfollowTags?: (tags, category?) => void;
```
Comment (idoshamun) [inline packages/shared/src/components/filters/CategoryButton.tsx:32]: Function arguments don't have types here

File: packages/shared/src/components/filters/TagButton.tsx [hunk 3]
```diff
@@ -72,32 +92,56 @@ const FollowTagButton = ({
 export interface TagButtonProps {
   tagItem: string;
   followedTags?: Array<string>;
+  blockedTags?: Array<string>;
   onFollowTags?: (tags, category?) => void;
   onUnfollowTags?: (tags, category?) => void;
+  onUnblockTags?: (tags) => void;
```
Comment (idoshamun) [inline packages/shared/src/components/filters/TagButton.tsx:98]: Functions here don't have types for their arguments

File: packages/shared/src/components/filters/TagCategoryDropdown.tsx [hunk 4]
```diff
@@ -12,13 +12,17 @@ import {
...
+  onUnblockTags,
 }: {
   tagCategory: TagCategory;
   followedTags?: Array<string>;
-  onFollowTags: (tags, category?) => void;
-  onUnfollowTags: (tags, category?) => void;
+  blockedTags?: Array<string>;
+  onFollowTags?: (tags, category?) => void;
+  onUnfollowTags?: (tags, category?) => void;
+  onUnblockTags?: (tags) => void;
```
Comment (idoshamun) [inline packages/shared/src/components/filters/TagCategoryDropdown.tsx:25]: Functions arguments don't have types

File: packages/shared/src/components/filters/TagItemList.tsx [hunk 5]
```diff
@@ -7,11 +7,19 @@ export default function TagItemList({
...
+  onUnblockTags,
   options,
 }: {
   tags: Tag[] | Array<string>;
   emptyText: string;
   followedTags?: Array<string>;
+  blockedTags?: Array<string>;
+  onFollowTags?: (tags, category?) => void;
+  onUnfollowTags?: (tags, category?) => void;
+  onUnblockTags?: (tags) => void;
```
Comment (idoshamun) [inline packages/shared/src/components/filters/TagItemList.tsx:22]: No types

**Review by idoshamun — CHANGES_REQUESTED** [review]
Please check my comments

**Comment by rebelchris** [comment]
@idoshamun Added types :)

**Review by idoshamun — APPROVED** [review]
Perfect!

**Pull request merged**